python/00scripts.md

```diff
@@ -0,0 +1,271 @@
+call whilst we’re just playing around.
+
+```python
+import sys
+import time
+
+def make_pizza(*toppings):
+    """Make a delicious pizza from the toppings."""
+    print('Making pizza...')
+    for topping in toppings:
+        print('Adding {0}'.format(topping))
+        time.sleep(1)
+    print('Done!')
+    return 'Pizza with toppings: {0}'.format(toppings)
+
+print('sys.argv:', sys.argv)
+# pizza = make_pizza('cheese', 'olives')
+```
+
+Then run it:
+
+```shell
+$ python pizzaiolo.py
+sys.argv: ['pizzaiolo.py']
+$ python pizzaiolo.py cheese broccoli
+sys.argv: ['pizzaiolo.py', 'cheese', 'broccoli']
+$ python pizzaiolo.py cheese broccoli --help --verbose
+sys.argv: ['pizzaiolo.py', 'cheese', 'broccoli', '--help', '--verbose']
+```
+
+Awesome! `sys.argv` is just a list with one value per argument (arguments on
+the command line are separate by spaces). The first value is always the name of
+the script that we run.
+
+So, let’s use the command line arguments in our script.
+
+```python
+# print('sys.argv:', sys.argv)
+toppings = sys.argv[1:]
+pizza = make_pizza(*toppings)
+```
+
+And then run it:
+
+```shell
+$ python pizzaiolo.py cheese broccoli
+Making pizza...
+Adding cheese
+Done!
+Adding broccoli
+Done!
+```
+
+Super cool. Now we could decide to add some flags that modify the behaviour of
+our program. We might like the `--help` flag to print a help message and exit,
+without actually making pizza, and a `--verbose` flag to enable printing more
+information.
+
+```python
+arguments = sys.argv[1:]
+
+if '--help' in arguments:
+    print('Make a pizza.')
+    print('Usage: pizzaiolo.py topping1 topping2 ...')
+    sys.exit()
+
+if '--verbose' in arguments:
+    verbose = True
+    # We don't want the flag passed as a topping!
+    arguments.remove('--verbose')
+else:
+    verbose = False
+
+if verbose:
+    print('About to call make_pizza')
+pizza = make_pizza(*arguments)
+if verbose:
+    print('Finished')
+```
+
+Let’s try running it.
+
+```shell
+$ python pizzaiolo.py cheese broccoli
+Making pizza...
+Adding cheese
+Done!
+Adding broccoli
+Done!
+$ python pizzaiolo.py cheese broccoli --help
+Make a pizza.
+Usage: pizzaiolo.py topping1 topping2 ...
+$ python pizzaiolo.py cheese broccoli --verbose
+About to call make_pizza
+Making pizza...
+Adding cheese
+Done!
+Adding broccoli
+Done!
+Finished
+```
+
+Great, everything seems to work.
+
+## argparse
+
+The [`argparse` module][argparse] comes as part of the Python standard library,
+and allows us to define what arguments our scripts accept much more easily
+than what we’ve shown. Under the hood, it just inspects `sys.argv` in exactly
+the same way as we’ve done, but it takes care of things like validation for us.
+
+```
+import argparse
+import sys
+import time
+
+
+def make_pizza(*toppings):
+    """Make a delicious pizza from the toppings."""
+    print('Making pizza...')
+    for topping in toppings:
+        print('Adding {0}'.format(topping))
```

Comment:
How about using f-strings here?

Reply:
oc, now that we are in 3.8 anyway. Replaced all `format`